modules/light-clients/08-wasm/types/migrate_contract.go

```diff
@@ -0,0 +1,34 @@
+package types
+
+import (
+	"bytes"
+
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// MigrateContract calls the migrate entry point on the contract with the given
+// migrateMsg. The contract must exist and the code hash must be found in the
+// store. If the code hash is the same as the current code hash, return nil.
+// This does not update the code hash in the client state.
+func (cs ClientState) MigrateContract(
+	ctx sdk.Context, cdc codec.BinaryCodec, clientID string,
+	clientStore storetypes.KVStore, newCodeHash, migrateMsg []byte,
+) error {
+	if !HasCodeHash(ctx, newCodeHash) {
+		return ErrWasmCodeHashNotFound
+	}
+
+	if bytes.Equal(cs.CodeHash, newCodeHash) {
+		return nil
+	}
```

Comment:
https://github.com/cosmos/ibc-go/issues/4925 explains that we should return early if code hashes are the same. I think we should error. I'd like to hear other's opinions on this

Reply:
gov should pay for their mistakes :money_with_wings:

Reply:
they'd pay either way right?

Reply:
Either works for me.

Reply:
turned into error